src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -235,6 +235,34 @@ You can use this event to run code on every page navigation, or only once ever:
   }, { once: true });
 </script>
 ```
+
+#### `astro:beforeload`
+
+An event that fires immediately after the new page replaces the old page. You can listen to this event on the `document`.
+
+This event is useful to restore any state on the DOM that needs to transfer over to the new page. If you are implementing dark mode support, this event can be used to restore state onto the new page like so:
+
+```astro
+<script>
+	const toggle = () => {
+		if(localStorage.darkMode) {
+			document.documentElement.dataset.dark = '';
+		}
+	};
+
+  // MPA nav
+	toggle();
+  // ViewTransitions nav
+	document.addEventListener('astro:beforeload', toggle);
+</script>
+<style is:global>
+	html[data-dark] body {
+		background: black;
+		color: white;
+	}
+</style>
```

Comment:
I think we can do some renaming here to clarify what’s happening, plus drop the `<style>` block which isn’t crucial to the example:

```suggestion
<script>
	const setDarkMode = () => {
		if (localStorage.darkMode) {
			document.documentElement.dataset.dark = '';
		}
	};

  // Runs on initial navigation
	setDarkMode();
  // Runs on view transitions navigation
	document.addEventListener('astro:beforeload', setDarkMode);
</script>
```